src/v/cluster/controller_backend.cc

```diff
@@ -1295,21 +1297,13 @@ ss::future<std::error_code> controller_backend::create_partition(
       log_revision,
       initial_replicas);
 
-    auto ec = co_await _shard_placement.prepare_create(ntp, log_revision);
-    if (ec) {
-        co_return ec;
-    }
-
     topic_configuration cfg;
     model::revision_id topic_rev;
     model::initial_revision_id remote_rev;
     {
         auto topic_md = _topics.local().get_topic_metadata_ref(
           model::topic_namespace_view(ntp));
-        if (!topic_md) {
-            // topic was already removed, do nothing
-            co_return errc::success;
-        }
+        vassert(topic_md, "topic metadata disappeared for {}", ntp);
```

Comment:
What prevents this vassert from being triggered?

Reply:
we are in the same scheduling segment as on line 974 where we make sure topic table has replica data for the NTP:
```
    std::optional<topic_table::partition_replicas_view> maybe_replicas_view
      = _topics.local().get_replicas_view(ntp);
    if (!maybe_replicas_view) {
        ...
        co_return ss::stop_iteration::yes;
    }
```